typo in smilesformat.cpp

hello

just tried to build latest master (5a9457ab01b473d1057ecd1dca107f4d49844c75) and getting this compile error:

    /home/balducci/tmp/install-us-d/openbabel-4950.d/openbabel-4958/src/formats/smilesformat.cpp:2869:14: error: cannot convert 'bool' to 'const char*' in return
           return false;

Things get fixed for me with:

diff -c src/formats/smilesformat.cpp.FIX_SMILESFORMAT_CPP src/formats/smilesformat.cpp
*** src/formats/smilesformat.cpp.FIX_SMILESFORMAT_CPP	Fri Nov 24 15:51:44 2017
--- src/formats/smilesformat.cpp	Fri Nov 24 15:51:44 2017
***************
*** 2866,2872 ****
    {
      // If no chiral neighbors were passed in, we're done
      if (chiral_neighbors.size() < 4)
!       return false;
      
      OBAtom *atom = node->GetAtom();
  
--- 2866,2872 ----
    {
      // If no chiral neighbors were passed in, we're done
      if (chiral_neighbors.size() < 4)
!       return (const char*)false;
      
      OBAtom *atom = node->GetAtom();


thanks

ciao
-gabriele

--- a/src/formats/smilesformat.cpp
+++ b/src/formats/smilesformat.cpp
@@ -2866,7 +2866,7 @@ namespace OpenBabel {
   {
     // If no chiral neighbors were passed in, we're done
     if (chiral_neighbors.size() < 4)
-      return false;
+      return (const char*)0;
     
     OBAtom *atom = node->GetAtom();
 

